File: 3114-latest-time-you-can-obtain-after-replacing-characters/3114-latest-time-you-can-obtain-after-replacing-characters.py
class Solution:
    def findLatestTime(self, s: str) -> str:
        output = ""
        if s[0] == "?" and s[1] == "?":
            output += "11"
        elif s[0] == "?" and s[1] > "1":
            output = output + "0" + s[1]
        elif s[0] == "?":
            output = output + "1" + s[1]
        elif s[0] == "0" and s[1] == "?":
            output += "09"
        elif s[0] == "1" and s[1] == "?":
            output += "11"
        else:
            output = output + s[0] + s[1]
        output += ":"
        if s[3] == "?":
            output += "5"
        else:
            output += s[3]
        if s[4] == "?":
            output += "9"
        else:
            output += s[4]
        return output
            
        

File: 3114-latest-time-you-can-obtain-after-replacing-characters/3114-latest-time-you-can-obtain-after-replacing-characters.java
class Solution {
    public String findLatestTime(String s) {
        String output = "";
        if(s.charAt(0) == '?' && s.charAt(1) == '?'){
            output += "11";
        } else if(s.charAt(0) == '?' && s.charAt(1) > '1'){
            output += '0' + String.valueOf(s.charAt(1));
        } else if(s.charAt(0) == '?'){
            output += '1' + String.valueOf(s.charAt(1));
        } else if(s.charAt(0) == '0' && s.charAt(1) == '?'){
            output += "09"; 
        } else if(s.charAt(0) == '1' && s.charAt(1) == '?') {
            output += "11";
        } else {
            output += String.valueOf(s.charAt(0)) + String.valueOf(s.charAt(1));
        }
        output += ":";
        if(s.charAt(3) == '?'){
            output += "5";
        } else {
            output += String.valueOf(s.charAt(3));
        }
        if(s.charAt(4) == '?'){
            output += "9";
        } else {
            output += String.valueOf(s.charAt(4));
        }
        return output;
    }
}